lib/iris/coords.py

```diff
@@ -1513,48 +1696,38 @@ def bounds(self):
 
         """
         bounds = None
-        if self._bounds is not None:
-            bounds = self._bounds
+        if self._bounds_dm is not None:
+            bounds = self._bounds_dm.data
         return bounds
 
     @bounds.setter
     def bounds(self, bounds):
-        if bounds is not None:
-            # Ensure the bounds are a compatible shape.
-            bounds = np.array(bounds, ndmin=2)
-            if self.shape != bounds.shape[:-1]:
-                raise ValueError(
-                    "The shape of the bounds array should be "
-                    "points.shape + (n_bounds,)")
-            # Checks for numeric and monotonic
-            if not np.issubdtype(bounds.dtype, np.number):
-                raise ValueError('The bounds array must be numeric.')
-
-            n_bounds = bounds.shape[-1]
-            n_points = bounds.shape[0]
-            if n_points > 1:
-
-                directions = set()
-                for b_index in range(n_bounds):
-                    monotonic, direction = iris.util.monotonic(
-                        bounds[:, b_index], strict=True, return_direction=True)
-                    if not monotonic:
-                        raise ValueError('The bounds array must be strictly '
-                                         'monotonic.')
-                    directions.add(direction)
-
-                if len(directions) != 1:
-                    raise ValueError('The direction of monotonicity must be '
-                                     'consistent across all bounds')
+        if bounds is None:
+            self._bounds_dm = None
+        else:
+            # Checks for appropriate values for the new bounds.
+            self._new_bounds_requirements(bounds)
 
             # Ensure the array is read-only.
             bounds.flags.writeable = False
-
-        self._bounds = bounds
+            self._bounds_dm.data = bounds
```

Comment:
If this coord currently has no bounds, `self._bounds_dm` will be `None` and this will raise and error.